Project architecture/MVVM.md

```diff
@@ -35,6 +34,39 @@ The ViewModel might sound powerful, but this is just a simple abstract class in
 However, the SingleLiveEvent implementation from Google has one major setback—it works only with one observer at a time. This issue is addressed and explained in detail in this [article](https://proandroiddev.com/livedata-with-single-events-2395dea972a8). The author of the article proposes an improved implementation, called LiveEvent, which covers the case of multiple observers. The source for implementation: [LiveEvent](https://github.com/hadilq/LiveEvent/blob/main/live-event/src/main/java/com/hadilq/liveevent/LiveEvent.kt).
 
+### Flows
+
+[Flows](https://developer.android.com/kotlin/flow) are conceptually a stream of data which flows in a pipe and can be computed asynchronously. On both ends of that pipe, there is a producer and a consumer running on coroutines (flows are built on top of coroutines). Suspend functions return only a single value and flows come in handy when we need to emit multiple values sequentially.
+
+### StateFlow and SharedFlow
+
+[StateFlow](https://developer.android.com/kotlin/flow/stateflow-and-sharedflow#stateflow) and [SharedFlow](https://developer.android.com/kotlin/flow/stateflow-and-sharedflow#sharedflow) are Flow APIs which we use to enable flows to optimally emit state updates and emit values to multiple consumers.
+
+By definition,
+
+> StateFlow is a state-holder observable flow that emits current and new state updates to its collectors.
+
+Essentially, StateFlow is very convenient for keeping our view states. It is a great fit for maintaining observable mutable states and handling live state updates.
+
+On the other hand, SharedFLow is a highly-configurable generalization of StateFlow. 
+
+> All methods of SharedFlow are thread-safe and can be safely invoked from concurrent coroutines without external synchronization.
+
+SharedFlow is the perfect choice when dealing with events. With SharedFlow, we avoid the trouble of handling resubscriptions (the cases when we want to do a certain action only once, such as showing toasts, Snackbar, navigating between fragments, etc.). SharedFlow behaves as a hot flow and it emits values to all consumers that collect from it.
+
+
+### LiveData vs Flow vs StateFlow vs SharedFlow
```

Comment:
Should LiveData, Flow etc be in a separate file?
Common Android? And just reference their use here.
This would go into a more extensive discussion about the restructuring handbook. All comments are welcomed.